library/modules/bigip_gtm_dns_listener.py

```diff
@@ -0,0 +1,836 @@
+        'advertise',
+        'description',
+        'sourcePort',
+        'translateAddress',
+        'translatePort',
+        'vlansDisabled',
+        'vlansEnabled',
+        'vlans',
+        'rules',
+        'autoLasthop',
+        'lastHopPool',
+        'fallbackPersistence',
+        'ipProtocol',
+        'mask',
+        'disabled',
+        'enabled'
+    ]
+
+    returnables = [
+        'description',
+        'disabled',
+        'enabled',
+        'ip_protocol',
+        'mask',
+        'address',
+        'port',
+        'advertise',
+        'auto_lasthop',
+        'translate_address',
+        'translate_port',
+        'fallback_persistence',
+        'source_port',
+        'vlans',
+        'last_hop_pool',
+        'rules',
+        'vlans_enabled',
+        'vlans_disabled'
+    ]
+
+    updatables = [
+        'description',
+        'disabled',
+        'enabled',
+        'mask',
+        'address',
+        'port',
+        'advertise',
+        'auto_lasthop',
+        'state',
+        'ip_protocol',
+        'fallback_persistence',
+        'translate_address',
+        'translate_port',
+        'source_port',
+        'vlans',
+        'last_hop_pool',
+        'rules',
+        'vlans_enabled',
+        'vlans_disabled'
+    ]
+
+    @property
+    def state(self):
+        if self._values['state'] == 'enabled':
+            return 'present'
+        return self._values['state']
+
+    @property
+    def enabled(self):
+        if self._values['enabled'] is None:
+            return None
+        return True
+
+    @property
+    def disabled(self):
+        if self._values['disabled'] is None:
+            return None
+        return True
+
+
+class ApiParameters(Parameters):
+    pass
+
+
+class ModuleParameters(Parameters):
+    @property
+    def description(self):
+        if self._values['description'] is None:
+            return None
+        elif self._values['description'] in ['none', '']:
+            return ''
+        return self._values['description']
+
+    @property
+    def address(self):
+        if self._values['address'] is None:
+            return None
+        return self._values['address']
+
+    @property
+    def port(self):
+        if self._values['port'] is None:
+            return None
+        return self._values['port']
+
+    @property
+    def ip_protocol(self):
+        if self._values['ip_protocol'] is None:
+            return None
+        return self._values['ip_protocol']
+
+    @property
+    def mask(self):
+        if self._values['mask'] is None:
+            return None
+        return self._values['mask']
+
+    @property
+    def translate_address(self):
+        if self._values['translate_address'] is None:
+            return None
+        return self._values['translate_address']
+
+    @property
+    def translate_port(self):
+        if self._values['translate_port'] is None:
+            return None
+        return self._values['translate_port']
```

Comment:
why are those here?

Reply:
I agree - since we are not transforming those parameters values, defining them here are unnecessary. I have removed all properties which do not require transformation from ModuleParameter class. 

This comment is addressed now. Let me know if you have any questions/concerns